src/actions/commands/actions.ts

```diff
@@ -1840,10 +1861,123 @@ class CommandShowCommandLine extends BaseCommand {
   }
 }
 
+@RegisterAction
+class CommandNavigateInCommandlineOrSearchMode extends BaseCommand {
+  modes = [ModeName.CommandlineInProgress, ModeName.SearchInProgressMode];
+  keys = [['<left>'], ['<right>']];
+  runsOnceForEveryCursor() {
+    return this.keysPressed[0] === '\n';
+  }
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    const key = this.keysPressed[0];
+    const searchState = vimState.globalState.searchState!;
+    const statusbartext =
+      searchState !== undefined && searchState!.searchString
+        ? searchState!.searchString
+        : vimState.currentCommandlineText;
+    if (key === '<right>') {
+      vimState.statusBarCursorCharacterPos = Math.min(
+        vimState.statusBarCursorCharacterPos + 1,
+        statusbartext.length
+      );
+    } else if (key === '<left>') {
+      vimState.statusBarCursorCharacterPos = Math.max(vimState.statusBarCursorCharacterPos - 1, 0);
+    }
+    return vimState;
+  }
+}
+@RegisterAction
+class CommandTabInCommandline extends BaseCommand {
+  modes = [ModeName.CommandlineInProgress];
+  keys = ['<tab>'];
+  runsOnceForEveryCursor() {
+    return this.keysPressed[0] === '\n';
+  }
+
+  private autoComplete(completionItems: any, vimState: VimState) {
+    if (
```

Comment:
I think this might be easier to read...

```
    if (commandLine.lastKeyPressed === '<tab>')
        if (commandLine.autoCompleteIndex < completionItems.length)
           commandLine.autoCompleteIndex += 1;
        else if (/ /g.test(vimState.currentCommandlineText)) {
            ...
    else
```

Reply:
I could do somthing like 
```
    if (commandLine.lastKeyPressed === '<tab>'){
      if (commandLine.autoCompleteIndex < completionItems.length) {
        commandLine.autoCompleteIndex += 1;
      }
    } else if (/ /g.test(vimState.currentCommandlineText)) {
      ...
    } else {
      ...
    }
```
but the way you rewrote it, it would not work due to it changing the behavior of the code.

Reply:
Yah, you're right. Nevermind.